hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/RecordListBuilder.java

```diff
@@ -253,19 +253,21 @@ private SingleTransactionRecordBuilderImpl doAddChild(
         }
 
         // Make sure we have not created so many that we have run out of slots.
-        final var childCount = childRecordBuilders.size();
-        final var consensusConfig = configuration.getConfigData(ConsensusConfig.class);
+        final int childCount = childRecordBuilders.size();
+        final ConsensusConfig consensusConfig = configuration.getConfigData(ConsensusConfig.class);
         if (childCount >= consensusConfig.handleMaxFollowingRecords()) {
             throw new HandleException(ResponseCodeEnum.MAX_CHILD_RECORDS_EXCEEDED);
         }
 
-        // The consensus timestamp of the first item in the child list is T+1, where T is the time of the user tx
-        final var parentConsensusTimestamp = userTxnRecordBuilder.consensusNow();
-        final var prevConsensusNow = childRecordBuilders.isEmpty()
-                ? userTxnRecordBuilder.consensusNow()
+        // The consensus timestamp of the first item in the child list is T+K (in nanoseconds),
```

Comment:
This is not what was explained to me when we talked about timestamps and I do not comfortable approving it. I think @tinker-michaelj needs to review the code.

Reply:
Understood.  The tests require this offset, so mono clearly _does this_, but the change should still be reviewed most carefully.

Reply:
We discussed this.
It turns out that Scheduled child transactions are very much treated like new top level transactions, instead of like children, in most situations, including this one.  The Scheduled transaction is offset an extra [max preceding transactions] because the scheduled transaction may, itself, have additional preceding transactions, which would (in mono) end up with the same nanosecond as other transactions if the scheduled (triggered) transaction doesn't offset this extra amount.

We might not handle the "scheduled transaction triggers a preceding transaction" situation in modular code in the preceding transaction dispatch ideally at the moment, but we don't think it's necessary to support in the first modular release (it just requires yet more special case handling in the preceding transaction logic if we do decide to support that edge case, however).